include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -543,16 +543,17 @@ __pattern_adjacent_find(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _R
     using _TagType = ::std::conditional_t<__is__or_semantic(), oneapi::dpl::__par_backend_hetero::__parallel_or_tag,
                                           oneapi::dpl::__par_backend_hetero::__parallel_find_forward_tag<_Range>>;
 
+    //ATTENTION!!! oneDPL supports SYCL buffer via a placeholder accessor; a 'subrange' cannot used here because
+    //getting an iterator for the placeholder accessor is incorrect on the host; so, oneDPL uses lazy-access views
+    //for range transformations.
+
 #if _ONEDPL_CPP20_RANGES_PRESENT
-    auto __first = std::ranges::begin(__rng);
     const auto __n = std::ranges::size(__rng);
-    auto __rng1 = std::ranges::subrange(__first, __first + __n - 1);
-    auto __rng2 = std::ranges::subrange(__first + 1, __first + __n);
+    auto __rng1 = std::ranges::take_view(__rng, __n - 1);
+    auto __rng2 = std::ranges::drop_view(__rng, 1);
```

Comment:
Can we use `take_view_simple` and `drop_view_simple` here to avoid issues with libstdc++10?

Reply:
Does libstdc++10 already support C++20?
In that case, it makes sense to try...

Reply:
Yes, it does support the majority of c++20 features. See https://en.cppreference.com/w/cpp/compiler_support/20.